**OS path**
- 학원 노트북용 --> C:/Users/Playdata/Playdata_anl/1st_Project/dataset/
- 개인 노트북용 --> C:/Users/Brian/Desktop/playdata/Playdata_anl/1st_Project/dataset/

In [1]:
import numpy as np
import pandas as pd

### 지하철 데이터 + 인구 데이터 + 기상 데이터 결합
- 세 개의 데이터 셋을 결합해서 분석에 사용할 최종 데이터 셋 **subway_pop_weather**을 생성할 것이다.

**세 개의 데이터 셋 파일 불러오기**

In [2]:
subway_final = pd.read_csv("C:/Users/Playdata/Playdata_anl/1st_Project/dataset/subway/subway_final.csv")
subway_final.drop(['Unnamed: 0'], axis = 1, inplace = True) # 첫 번째 컬럼 제거(의미 없는 컬럼임)

pop_final = pd.read_csv("C:/Users/Playdata/Playdata_anl/1st_Project/dataset/population/total_final/pop_trans_final.csv")
pop_final.drop(['Unnamed: 0'], axis = 1, inplace = True) # 첫 번째 컬럼 제거(의미 없는 컬럼임)

weather_final = pd.read_csv("C:/Users/Playdata/Playdata_anl/1st_Project/dataset/weather/all_features_weather.csv", encoding = 'cp949')
weather_final.drop(['Unnamed: 0'], axis = 1, inplace = True) # 첫 번째 컬럼 제거(의미 없는 컬럼임)

In [3]:
subway_final.shape

(602304, 43)

In [4]:
weather_final.shape

(27375, 165)

In [5]:
pop_final.shape

(458109, 161)

**Check-Up!!**
- 여기서 한 가지 더 확인해봐야 할 부분이 있다.
    - **지하철 데이터 셋의 "gu" 변수**와 **기상 데이터 셋의 "구" 변수**가 **일치하는지 비교**해보아야 한다.
    - 앞서 subway_weather_pop_final 데이터 셋을 한 번 생성해보았을 때, **지하철 데이터 셋의 "gu" 변수**와 **기상 데이터 셋의 "구" 변수**가 **불일치해서 Null 값이 발생한 것을 확인**했다.
        - 따라서 앞에서(다른 주피터 파일에서) **subway_final 데이터 셋을 만들 때, "강남"을 "강남구"로 변경**해주었다.
    - 변경해 준 뒤, 각 데이터 셋의 "gu", "구" 변수가 일치하는지 다시 한 번 확인해보겠다.

In [10]:
comp1 = pd.DataFrame(set(subway_final['gu']))
comp1.rename(columns = {0:'gu_subway'}, inplace = True)

comp2 = pd.DataFrame(set(weather_final['구']))
comp2.rename(columns = {0:'gu_weather'}, inplace = True)

In [11]:
comp1

gu_subway
0       영등포구
1        도봉구
2        금천구
3        종로구
4        강동구
5        강북구
6   경기도 의정부시
7        송파구
8        부천시
9        용산구
10       구로구
11       성북구
12        중구
13       강서구
14       동작구
15       광진구
16       강남구
17   고양시 덕양구
18       광명시
19       성동구
20       부평구
21       관악구
22       중랑구
23      동대문구
24       마포구
25       서초구
26       성남시
27       노원구
28       양천구
29       은평구
30      서대문구

In [12]:
comp2

gu_weather
0         도봉구
1         금천구
2         은평구
3         종로구
4         강동구
5         강북구
6         송파구
7         용산구
8         구로구
9         성북구
10         중구
11        강서구
12        동작구
13        광진구
14        강남구
15        성동구
16        관악구
17        중랑구
18       동대문구
19        마포구
20        서초구
21        노원구
22        양천구
23       영등포구
24       서대문구

In [14]:
comp1.drop([6, 8, 17, 18, 20, 26], axis = 0, inplace = True)

In [15]:
len(comp1), len(comp2)

(25, 25)

In [16]:
comp1.sort_values(by = 'gu_subway', inplace = True)
comp2.sort_values(by = 'gu_weather', inplace = True)

comp = pd.concat([comp1, comp2], axis = 1)
comp

gu_subway gu_weather
0       영등포구        도봉구
1        도봉구        금천구
2        금천구        은평구
3        종로구        종로구
4        강동구        강동구
5        강북구        강북구
6        NaN        송파구
7        송파구        용산구
8        NaN        구로구
9        용산구        성북구
10       구로구         중구
11       성북구        강서구
12        중구        동작구
13       강서구        광진구
14       동작구        강남구
15       광진구        성동구
16       강남구        관악구
17       NaN        중랑구
18       NaN       동대문구
19       성동구        마포구
20       NaN        서초구
21       관악구        노원구
22       중랑구        양천구
23      동대문구       영등포구
24       마포구       서대문구
25       서초구        NaN
27       노원구        NaN
28       양천구        NaN
29       은평구        NaN
30      서대문구        NaN

- 다시 한 번 두 개의 데이터 셋에 gu 변수를 비교해보니, **일치함을 확인**할 수 있다!!

**1. subway_final + weather_final (데이터 셋 결합)**
- **weather_join_key 변수를 조인키로 사용**하여 **left join**
- 두 개의 데이터 셋이 결합된 **subway_weather_final 데이터 셋을 생성!!**

In [17]:
subway_weather_final = pd.merge(subway_final, weather_final, how = 'left', on = 'weather_join_key')
subway_weather_final

date line  station_num station_name boarding  05~06  06~07  \
0       2017-01-01  1호선          150          서울역       승차    470    286   
1       2017-01-01  1호선          150          서울역       하차    278    880   
2       2017-01-02  1호선          150          서울역       하차    356   2366   
3       2017-01-02  1호선          150          서울역       승차    411    536   
4       2017-01-03  1호선          150          서울역       하차    303   1719   
...            ...  ...          ...          ...      ...    ...    ...   
602299  2019-12-26  6호선         2615          연신내       승차      0      0   
602300  2019-12-27  6호선         2615          연신내       승차      0      0   
602301  2019-12-27  6호선         2615          연신내       하차      0      0   
602302  2019-12-28  6호선         2615          연신내       하차      0      0   
602303  2019-12-28  6호선         2615          연신내       승차      0      0   

        07~08  08~09  09~10  ...  눈내림여부(일별)_14  눈내림여부(일별)_15  눈내림여부(일별)_16  \
0         397    786   1421  ...           0.0           0.0           0.0   
1         859    964   1407  ...           0.0           0.0           0.0   
2        4852   9813   4784  ...           0.0           0.0           0.0   
3        2297   3665   2103  ...           0.0           0.0           0.0   
4        4452  10406   4863  ...           0.0           0.0           0.0   
...       ...    ...    ...  ...           ...           ...           ...   
602299      0      0      0  ...           0.0           0.0           0.0   
602300      0      0      0  ...           0.0           0.0           0.0   
602301      0      0      0  ...           0.0           0.0           0.0   
602302      0      0      0  ...           0.0           0.0           0.0   
602303      0      0      0  ...           0.0           0.0           0.0   

        눈내림여부(일별)_17  눈내림여부(일별)_18  눈내림여부(일별)_19  눈내림여부(일별)_20  눈내림여부(일별)_21  \
0                0.0           0.0           0.0           0.0           0.0   
1                0.0           0.0           0.0           0.0           0.0   
2                0.0           0.0           0.0           0.0           0.0   
3                0.0           0.0           0.0           0.0           0.0   
4                0.0           0.0           0.0           0.0           0.0   
...              ...           ...           ...           ...           ...   
602299           0.0           0.0           0.0           0.0           0.0   
602300           0.0           0.0           0.0           0.0           0.0   
602301           0.0           0.0           0.0           0.0           0.0   
602302           0.0           0.0           0.0           0.0           0.0   
602303           0.0           0.0           0.0           0.0           0.0   

        눈내림여부(일별)_22  눈내림여부(일별)_23  
0                0.0           0.0  
1                0.0           0.0  
2                0.0           0.0  
3                0.0           0.0  
4                0.0           0.0  
...              ...           ...  
602299           0.0           0.0  
602300           0.0           0.0  
602301           0.0           0.0  
602302           0.0           0.0  
602303           0.0           0.0  

[602304 rows x 207 columns]

In [18]:
len(subway_final)

602304

- 위 결과를 보면, left join이 잘 이루어진 것을 확인할 수 있다.

In [41]:
subway_weather_final.isnull().sum()

date                0
line                0
station_num         0
station_name        0
boarding            0
                ...  
눈내림여부(일별)_19    41610
눈내림여부(일별)_20    41610
눈내림여부(일별)_21    41610
눈내림여부(일별)_22    41610
눈내림여부(일별)_23    41610
Length: 207, dtype: int64

- 위의 **41610개의 Null 값들**은 **'경기도', '인천광역시'에 속한 데이터의 개수**이므로 **무시해도 무방**하다!!
    - 이 내용은 코드 맨 마지막 쪽에 자세하게 설명해놓았다.

**2. subway_weather_final + pop_final (데이터 셋 결합)**
- **pop_join_key 변수를 조인키로 사용**하여 **left join**
- 두 개의 데이터 셋이 결합된 **subway_weather_pop_final 데이터 셋을 생성!!**

In [21]:
subway_weather_pop_final = pd.merge(subway_weather_final, pop_final, how = 'left', on = 'pop_join_key')
subway_weather_pop_final

date line  station_num station_name boarding  05~06  06~07  \
0       2017-01-01  1호선          150          서울역       승차    470    286   
1       2017-01-01  1호선          150          서울역       하차    278    880   
2       2017-01-02  1호선          150          서울역       하차    356   2366   
3       2017-01-02  1호선          150          서울역       승차    411    536   
4       2017-01-03  1호선          150          서울역       하차    303   1719   
...            ...  ...          ...          ...      ...    ...    ...   
602299  2019-12-26  6호선         2615          연신내       승차      0      0   
602300  2019-12-27  6호선         2615          연신내       승차      0      0   
602301  2019-12-27  6호선         2615          연신내       하차      0      0   
602302  2019-12-28  6호선         2615          연신내       하차      0      0   
602303  2019-12-28  6호선         2615          연신내       승차      0      0   

        07~08  08~09  09~10  ...    total_19    total_20    total_21  \
0         397    786   1421  ...  13965.0479  14464.2979  15162.6506   
1         859    964   1407  ...  13965.0479  14464.2979  15162.6506   
2        4852   9813   4784  ...  25181.3513  26227.9772  27251.8970   
3        2297   3665   2103  ...  25181.3513  26227.9772  27251.8970   
4        4452  10406   4863  ...  52916.4980  52575.2426  52218.9502   
...       ...    ...    ...  ...         ...         ...         ...   
602299      0      0      0  ...  19112.9971  19477.1296  19419.9693   
602300      0      0      0  ...  40842.2139  39608.0210  39002.3523   
602301      0      0      0  ...  40842.2139  39608.0210  39002.3523   
602302      0      0      0  ...  23148.4284  23315.2732  23715.6562   
602303      0      0      0  ...  23148.4284  23315.2732  23715.6562   

          total_22    total_23     total_5     total_6     total_7  \
0       16336.8151  16005.3908  16668.3599  16699.1654  15776.5716   
1       16336.8151  16005.3908  16668.3599  16699.1654  15776.5716   
2       27774.0245  32401.8704  32491.9088  31453.9491  29285.2527   
3       27774.0245  32401.8704  32491.9088  31453.9491  29285.2527   
4       52409.0778  47157.2329  47120.9495  47685.4421  47704.0643   
...            ...         ...         ...         ...         ...   
602299  20006.9136  19744.3681  18455.1459  17957.4967  19155.8356   
602300  38903.4575  36007.7910  32948.3544  34037.7242  39395.0839   
602301  38903.4575  36007.7910  32948.3544  34037.7242  39395.0839   
602302  24213.1422  24002.7606  24079.6910  24250.7596  26299.7750   
602303  24213.1422  24002.7606  24079.6910  24250.7596  26299.7750   

           total_8     total_9  
0       14370.7991  13410.9693  
1       14370.7991  13410.9693  
2       25878.3191  24701.9018  
3       25878.3191  24701.9018  
4       45729.1596  44860.2133  
...            ...         ...  
602299  16839.4103  15768.3616  
602300  38533.1286  37710.2694  
602301  38533.1286  37710.2694  
602302  24654.0285  23424.4790  
602303  24654.0285  23424.4790  

[602304 rows x 367 columns]

In [22]:
len(subway_weather_final)

602304

In [23]:
# str "not" contains
check1 = subway_weather_pop_final[(~subway_weather_pop_final['address'].str.contains('서울특별시'))] # '서울특별시' 제외
check2 = check1[~check1['address'].str.contains('경기도')] # '경기도' 제외
check3 = check2[~check2['address'].str.contains('인천광역시')] # '인천광역시' 제외
check3

Empty DataFrame
Columns: [date, line, station_num, station_name, boarding, 05~06, 06~07, 07~08, 08~09, 09~10, 10~11, 11~12, 12~13, 13~14, 14~15, 15~16, 16~17, 17~18, 18~19, 19~20, 20~21, 21~22, 22~23, 23~24, 00~01, total, type, area, gu, address, transfer, transfer_num, entrance, gate, nsi_cnt, hsi_cnt, msi_cnt, uvi_cnt, hti_cnt, mti_cnt, dong, pop_join_key, weather_join_key, 지점명, 주소, 구, 날짜, 기온_0, 기온_5, 기온_6, 기온_7, 기온_8, 기온_9, 기온_10, 기온_11, 기온_12, 기온_13, 기온_14, 기온_15, 기온_16, 기온_17, 기온_18, 기온_19, 기온_20, 기온_21, 기온_22, 기온_23, 풍속(m/s)_0, 풍속(m/s)_5, 풍속(m/s)_6, 풍속(m/s)_7, 풍속(m/s)_8, 풍속(m/s)_9, 풍속(m/s)_10, 풍속(m/s)_11, 풍속(m/s)_12, 풍속(m/s)_13, 풍속(m/s)_14, 풍속(m/s)_15, 풍속(m/s)_16, 풍속(m/s)_17, 풍속(m/s)_18, 풍속(m/s)_19, 풍속(m/s)_20, 풍속(m/s)_21, 풍속(m/s)_22, 풍속(m/s)_23, 습도(%)_0, 습도(%)_5, 습도(%)_6, 습도(%)_7, 습도(%)_8, 습도(%)_9, 습도(%)_10, 습도(%)_11, 습도(%)_12, 습도(%)_13, 습도(%)_14, 습도(%)_15, 습도(%)_16, ...]
Index: []

[0 rows x 367 columns]

- 위 결과를 보면, **subway_weather_pop_final 데이터 셋이 '서울특별시', '경기도', '인천광역시'로 이루어진 것을 알 수 있다.**

In [24]:
subway_weather_pop_final[subway_weather_pop_final['address'].str.contains('서울특별시')].isnull().sum()

date               0
line               0
station_num        0
station_name       0
boarding           0
                ... 
total_5         6144
total_6         6144
total_7         6144
total_8         6144
total_9         6144
Length: 367, dtype: int64

In [25]:
null_check = subway_weather_pop_final[subway_weather_pop_final['address'].str.contains('서울특별시')].isnull().sum()
df_null_check = pd.DataFrame(null_check)

In [26]:
df_null_check[df_null_check[0] != 0]

0
transfer     352590
gate             38
below_20_0     6144
below_20_10    6144
below_20_11    6144
...             ...
total_5        6144
total_6        6144
total_7        6144
total_8        6144
total_9        6144

[162 rows x 1 columns]

- Left join 후에 존재하는 Null 값들에 대한 고찰...
    - **얼핏 보면 left join이 잘 된 것 같지만, 그래도 다시 한 번 check!!**

In [27]:
cc = subway_weather_pop_final[subway_weather_pop_final['address'].str.contains('서울특별시')]
cc_list = list(set(cc['station_num'])) # 258개
len(cc_list)

258

In [28]:
tt = cc[cc['total_5'].isnull()]
tt_list = list(set(tt['station_num'])) # 256개
len(tt_list)

256

- **"258 - 256 = 2"** 이므로, **차이나는 2개의 지하철 역이 어디인지 파악**해보자.

In [29]:
for i in tt_list:
    if i in cc_list:
        cc_list.pop(cc_list.index(i))
        
cc_list

[2615, 321]

- 확인해보니 **station_num 변수가 2615, 321인 부분에서 차이가 발생**했다.

In [30]:
cc[(cc['station_num'] == 2615) | (cc['station_num'] == 321)]

date line  station_num station_name boarding  05~06  06~07  \
602250  2019-12-01  3호선          321          충무로       승차      0      0   
602251  2019-12-01  3호선          321          충무로       하차      0      0   
602252  2019-12-04  3호선          321          충무로       승차      0      0   
602253  2019-12-04  3호선          321          충무로       하차      0      0   
602254  2019-12-05  3호선          321          충무로       승차      0      0   
602255  2019-12-05  3호선          321          충무로       하차      0      0   
602256  2019-12-06  3호선          321          충무로       승차      0      0   
602257  2019-12-06  3호선          321          충무로       하차      0      0   
602258  2019-12-07  3호선          321          충무로       승차      0      0   
602259  2019-12-07  3호선          321          충무로       하차      0      0   
602260  2019-12-09  3호선          321          충무로       승차      0      0   
602261  2019-12-09  3호선          321          충무로       하차      0      0   
602262  2019-12-11  3호선          321          충무로       하차      0      0   
602263  2019-12-11  3호선          321          충무로       승차      0      0   
602264  2019-12-12  3호선          321          충무로       승차      0      0   
602265  2019-12-12  3호선          321          충무로       하차      0      0   
602266  2019-12-13  3호선          321          충무로       하차      0      0   
602267  2019-12-13  3호선          321          충무로       승차      0      0   
602268  2019-12-14  3호선          321          충무로       승차      0      0   
602269  2019-12-14  3호선          321          충무로       하차      0      0   
602270  2019-12-16  3호선          321          충무로       하차      0      0   
602271  2019-12-16  3호선          321          충무로       승차      0      0   
602272  2019-12-17  3호선          321          충무로       승차      0      0   
602273  2019-12-17  3호선          321          충무로       하차      0      0   
602274  2019-12-20  3호선          321          충무로       하차      0      0   
602275  2019-12-20  3호선          321          충무로       승차      0      0   
602276  2019-12-21  3호선          321          충무로       승차      0      0   
602277  2019-12-21  3호선          321          충무로       하차      0      0   
602278  2019-12-22  3호선          321          충무로       승차      0      0   
602279  2019-12-22  3호선          321          충무로       하차      0      0   
602280  2019-12-25  3호선          321          충무로       승차      0      0   
602281  2019-12-25  3호선          321          충무로       하차      0      0   
602282  2019-12-26  3호선          321          충무로       승차      0      0   
602283  2019-12-26  3호선          321          충무로       하차      0      0   
602284  2019-12-30  3호선          321          충무로       승차      0      0   
602285  2019-12-30  3호선          321          충무로       하차      0      0   
602286  2019-12-31  3호선          321          충무로       승차      0      0   
602287  2019-12-31  3호선          321          충무로       하차      0      0   
602288  2019-12-15  6호선         2615          연신내       승차      0      0   
602289  2019-12-15  6호선         2615          연신내       하차      0      0   
602290  2019-12-16  6호선         2615          연신내       승차      0      0   
602291  2019-12-16  6호선         2615          연신내       하차      0      0   
602292  2019-12-17  6호선         2615          연신내       승차      0      0   
602293  2019-12-17  6호선         2615          연신내       하차      0      0   
602294  2019-12-22  6호선         2615          연신내       하차      0      0   
602295  2019-12-22  6호선         2615          연신내       승차      0      0   
602296  2019-12-23  6호선         2615          연신내       승차      0      0   
602297  2019-12-23  6호선         2615          연신내       하차      0      0   
602298  2019-12-26  6호선         2615          연신내       하차      0      0   
602299  2019-12-26  6호선         2615          연신내       승차      0      0   
602300  2019-12-27  6호선         2615          연신내       승차      0      0   
602301  2019-12-27  6호선         2615          연신내       하차    

- 확인해보니, **3호선 "충무로"역**과 **6호선 "연신내"역**에서 **차이가 발생**한 것이었다.
    - 3호선 "충무로"역과 6호선 "연신내"역의 gate 변수가 Null 값이 존재했는데, **해당 역의 gate는 미존재하기 때문에 당연한 결과**이다.

In [31]:
len(cc[(cc['station_num'] == 2615) | (cc['station_num'] == 321)]['gate'])

54

In [32]:
cc[(cc['station_num'] == 2615) | (cc['station_num'] == 321)]['gate'].isnull().sum()

38

In [33]:
subway_weather_pop_final[subway_weather_pop_final['address'].str.contains('서울특별시')]['gate'].isnull().sum()

38

- 이제 **분석 대상인 "서울특별시"에 해당하는 데이터들만 추출**해서 내보내주도록 하겠다.

In [34]:
subway_weather_pop_final2 = subway_weather_pop_final[subway_weather_pop_final['address'].str.contains('서울특별시')]
subway_weather_pop_final2

date line  station_num station_name boarding  05~06  06~07  \
0       2017-01-01  1호선          150          서울역       승차    470    286   
1       2017-01-01  1호선          150          서울역       하차    278    880   
2       2017-01-02  1호선          150          서울역       하차    356   2366   
3       2017-01-02  1호선          150          서울역       승차    411    536   
4       2017-01-03  1호선          150          서울역       하차    303   1719   
...            ...  ...          ...          ...      ...    ...    ...   
602299  2019-12-26  6호선         2615          연신내       승차      0      0   
602300  2019-12-27  6호선         2615          연신내       승차      0      0   
602301  2019-12-27  6호선         2615          연신내       하차      0      0   
602302  2019-12-28  6호선         2615          연신내       하차      0      0   
602303  2019-12-28  6호선         2615          연신내       승차      0      0   

        07~08  08~09  09~10  ...    total_19    total_20    total_21  \
0         397    786   1421  ...  13965.0479  14464.2979  15162.6506   
1         859    964   1407  ...  13965.0479  14464.2979  15162.6506   
2        4852   9813   4784  ...  25181.3513  26227.9772  27251.8970   
3        2297   3665   2103  ...  25181.3513  26227.9772  27251.8970   
4        4452  10406   4863  ...  52916.4980  52575.2426  52218.9502   
...       ...    ...    ...  ...         ...         ...         ...   
602299      0      0      0  ...  19112.9971  19477.1296  19419.9693   
602300      0      0      0  ...  40842.2139  39608.0210  39002.3523   
602301      0      0      0  ...  40842.2139  39608.0210  39002.3523   
602302      0      0      0  ...  23148.4284  23315.2732  23715.6562   
602303      0      0      0  ...  23148.4284  23315.2732  23715.6562   

          total_22    total_23     total_5     total_6     total_7  \
0       16336.8151  16005.3908  16668.3599  16699.1654  15776.5716   
1       16336.8151  16005.3908  16668.3599  16699.1654  15776.5716   
2       27774.0245  32401.8704  32491.9088  31453.9491  29285.2527   
3       27774.0245  32401.8704  32491.9088  31453.9491  29285.2527   
4       52409.0778  47157.2329  47120.9495  47685.4421  47704.0643   
...            ...         ...         ...         ...         ...   
602299  20006.9136  19744.3681  18455.1459  17957.4967  19155.8356   
602300  38903.4575  36007.7910  32948.3544  34037.7242  39395.0839   
602301  38903.4575  36007.7910  32948.3544  34037.7242  39395.0839   
602302  24213.1422  24002.7606  24079.6910  24250.7596  26299.7750   
602303  24213.1422  24002.7606  24079.6910  24250.7596  26299.7750   

           total_8     total_9  
0       14370.7991  13410.9693  
1       14370.7991  13410.9693  
2       25878.3191  24701.9018  
3       25878.3191  24701.9018  
4       45729.1596  44860.2133  
...            ...         ...  
602299  16839.4103  15768.3616  
602300  38533.1286  37710.2694  
602301  38533.1286  37710.2694  
602302  24654.0285  23424.4790  
602303  24654.0285  23424.4790  

[560694 rows x 367 columns]

- 마지막으로 기상 데이터에 존재했던 Null 값 문제들이, "강남"을 "강남구"로 수정해주면서 잘 해결되었는지 확인해보자!! (제발..ㅜㅜ)

In [40]:
# "원래 데이터 개수 - '서울특별시' 데이터 개수" = "다른 지역의 데이터 개수('경기도', '인천광역시')"
# 이 부분이 위 쪽에서 subway_weather_final 데이터 셋의 Null 값들은 무시해도 무방하다는 내용의 근거임
602304 - 560694

41610

In [42]:
subway_weather_pop_final2[subway_weather_pop_final2['구'].isnull()]

Empty DataFrame
Columns: [date, line, station_num, station_name, boarding, 05~06, 06~07, 07~08, 08~09, 09~10, 10~11, 11~12, 12~13, 13~14, 14~15, 15~16, 16~17, 17~18, 18~19, 19~20, 20~21, 21~22, 22~23, 23~24, 00~01, total, type, area, gu, address, transfer, transfer_num, entrance, gate, nsi_cnt, hsi_cnt, msi_cnt, uvi_cnt, hti_cnt, mti_cnt, dong, pop_join_key, weather_join_key, 지점명, 주소, 구, 날짜, 기온_0, 기온_5, 기온_6, 기온_7, 기온_8, 기온_9, 기온_10, 기온_11, 기온_12, 기온_13, 기온_14, 기온_15, 기온_16, 기온_17, 기온_18, 기온_19, 기온_20, 기온_21, 기온_22, 기온_23, 풍속(m/s)_0, 풍속(m/s)_5, 풍속(m/s)_6, 풍속(m/s)_7, 풍속(m/s)_8, 풍속(m/s)_9, 풍속(m/s)_10, 풍속(m/s)_11, 풍속(m/s)_12, 풍속(m/s)_13, 풍속(m/s)_14, 풍속(m/s)_15, 풍속(m/s)_16, 풍속(m/s)_17, 풍속(m/s)_18, 풍속(m/s)_19, 풍속(m/s)_20, 풍속(m/s)_21, 풍속(m/s)_22, 풍속(m/s)_23, 습도(%)_0, 습도(%)_5, 습도(%)_6, 습도(%)_7, 습도(%)_8, 습도(%)_9, 습도(%)_10, 습도(%)_11, 습도(%)_12, 습도(%)_13, 습도(%)_14, 습도(%)_15, 습도(%)_16, ...]
Index: []

[0 rows x 367 columns]

- subway_weather_pop_final2 데이터 셋의 **'구' 변수(기상 데이터 부분)를 확인해보면, Null 값들이 존재하지 않는다!! 유레카!!!**

In [44]:
weather_null_check = subway_weather_pop_final2.isnull().sum()
df_weather_null_check = pd.DataFrame(weather_null_check)
df_weather_null_check.rename(columns = {0:'null_value'}, inplace = True)
df_weather_null_check[df_weather_null_check['null_value'] != 0]

null_value
transfer         352590
gate                 38
below_20_0         6144
below_20_10        6144
below_20_11        6144
...                 ...
total_5            6144
total_6            6144
total_7            6144
total_8            6144
total_9            6144

[162 rows x 1 columns]

### 최종 결합된 데이터 셋을 csv 파일로 내보내기

In [45]:
subway_weather_pop_final2.to_csv("C:/Users/Playdata/Playdata_anl/1st_Project/dataset/subway_weather_pop_final.csv")